include/config/config_safeguards.h

```diff
@@ -0,0 +1,146 @@
+#pragma once
+
+/*****************************
+ * COMPATIBILITY SAFE GUARDS *
+ *****************************/
+
+/**
+ * NOTE: Only mess with these if you know what you're doing!
+ * These are put in place to insure that connected functionality works as intended.
+ */
+
+
+/*****************
+ * config_graphics
+ */
+
+#ifndef F3DZEX_GBI_2
+    #undef OBJECTS_REJ // OBJECTS_REJ requires f3dzex.
+#endif // !F3DZEX_GBI_2
+
+#ifndef F3DEX_GBI_SHARED
+    #undef OBJECTS_REJ // Non F3DEX-based ucodes do NOT support ucode switching.
+#endif // !F3DEX_GBI_SHARED
+
+#ifdef OBJECTS_REJ
+// Enable required ucodes.
+    #define F3DEX2_REJ_GBI
+    #define F3DLX2_REJ_GBI
+#endif // OBJECTS_REJ
+
+
+/*****************
+ * config_debug
+ */
+
+#ifdef PUPPYPRINT_DEBUG
+    #ifndef PUPPYPRINT
+        #define PUPPYPRINT
+    #endif
+#endif
+
+#ifdef COMPLETE_SAVE_FILE
+    #define UNLOCK_ALL
```

Comment:
UNLOCK_ALL can potentially be redefined here, which I believe causes a compiler warning.